src/ProxyClient/Fastly.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient;
+
+use FOS\HttpCache\ProxyClient\Invalidation\ClearCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\PurgeCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\RefreshCapable;
+use FOS\HttpCache\ProxyClient\Invalidation\TagCapable;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Fastly HTTP cache invalidator.
+ *
+ * Additional constructor options:
+ * - service_identifier    Identifier for your Fastly service account.
+ * - authentication_token  Token for authentication against Fastly APIs.
+ *                         For full capabilities (incl ClearCapable) you'll need one with Fastly Engineer permissions.
+ * - soft_purge            Boolean for doing soft purges or not on tag invalidation and url purging, default true.
+ *                         Soft purges expires cache instead of hard purge, and allow grace/stale handling.
+ *
+ * @link https://docs.fastly.com/api/purge Fastly Purge API documentation.
+ * @author Simone Fumagalli <simone.fumagalli@musement.com>
+ * @author André Rømcke <andre_gpt@msn.com>
+ */
+class Fastly extends HttpProxyClient implements ClearCapable, PurgeCapable, RefreshCapable, TagCapable
+{
+    /**
+     * @internal
+     */
+    const HTTP_METHOD_PURGE = 'PURGE';
+
+    /**
+     * @internal
+     * @see https://docs.fastly.com/api/purge#purge_db35b293f8a724717fcf25628d713583 Fastly's limit on batch tag purges.
+     */
+    const TAG_BATCH_PURGE_LIMIT = 256;
+
+    /**
+     * @internal
+     * @see https://docs.fastly.com/api/purge Base url endpoint used on anything but url PURGE/GET/HEAD.
+     */
+    const API_ENDPOINT = 'https://api.fastly.com';
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see https://docs.fastly.com/api/purge#purge_db35b293f8a724717fcf25628d713583
+     */
+    public function invalidateTags(array $tags)
+    {
+        $headers = [
+            'Fastly-Key' => $this->options['authentication_token'],
+            'Accept' => 'application/json',
+        ];
+
+        if (true === $this->options['soft_purge']) {
+            $headers['Fastly-Soft-Purge'] = 1;
+        }
+
+        // Split tag invalidations across several requests within Fastly's tag batch invalidations limits.
+        foreach (\array_chunk($tags, self::TAG_BATCH_PURGE_LIMIT) as $tagChunk) {
+            $this->queueRequest(
+                Request::METHOD_POST,
+                sprintf(self::API_ENDPOINT . '/service/%s/purge', $this->options['service_identifier']),
+                $headers + [
+                    // Note: Can be changed to rather use json payload if queueRequest is changed to allow passing body
```

Comment:
did we discuss that? i think we could add an optional body argument in HttpProxyClient::queueRequest, or not?

would be less fragile than the header and we could simply use the TAG_BATCH_PURGE_LIMIT constant without further worries.

Reply:
done